core/src/main/java/tc/oc/pgm/spawner/SpawnerModule.java

```diff
@@ -49,58 +54,82 @@ public SpawnerModule parse(MapFactory factory, Logger logger, Document doc)
-      for (Element element :
+      for (Element spawnerEl :
           XMLUtils.flattenElements(doc.getRootElement(), "spawners", "spawner")) {
-        Region spawnRegion = regionParser.parseRequiredRegionProperty(element, "spawn-region");
-        Region playerRegion = regionParser.parseRequiredRegionProperty(element, "player-region");
-        Attribute delayAttr = element.getAttribute("delay");
-        Attribute minDelayAttr = element.getAttribute("min-delay");
-        Attribute maxDelayAttr = element.getAttribute("max-delay");
+        Region spawnRegion = regionParser.parseRequiredRegionProperty(spawnerEl, "spawn-region");
+        Region playerRegion = regionParser.parseRequiredRegionProperty(spawnerEl, "player-region");
+        Attribute delayAttr = spawnerEl.getAttribute("delay");
+        Attribute minDelayAttr = spawnerEl.getAttribute("min-delay");
+        Attribute maxDelayAttr = spawnerEl.getAttribute("max-delay");
 
         if ((minDelayAttr != null || maxDelayAttr != null) && delayAttr != null) {
           throw new InvalidXMLException(
-              "Attribute 'minDelay' and 'maxDelay' cannot be combined with 'delay'", element);
+              "Attribute 'minDelay' and 'maxDelay' cannot be combined with 'delay'", spawnerEl);
         }
 
         Duration delay = XMLUtils.parseDuration(delayAttr, Duration.ofSeconds(10));
         Duration minDelay = XMLUtils.parseDuration(minDelayAttr, delay);
         Duration maxDelay = XMLUtils.parseDuration(maxDelayAttr, delay);
 
         if (maxDelay.compareTo(minDelay) <= 0 && minDelayAttr != null && maxDelayAttr != null) {
-          throw new InvalidXMLException("Max-delay must be longer than min-delay", element);
+          throw new InvalidXMLException("Max-delay must be longer than min-delay", spawnerEl);
         }
 
         int maxEntities =
             XMLUtils.parseNumber(
-                element.getAttribute("max-entities"), Integer.class, Integer.MAX_VALUE);
+                spawnerEl.getAttribute("max-entities"), Integer.class, Integer.MAX_VALUE);
         Filter playerFilter =
-            filterParser.parseFilterProperty(element, "filter", StaticFilter.ALLOW);
+            filterParser.parseFilterProperty(spawnerEl, "filter", StaticFilter.ALLOW);
 
         List<Spawnable> objects = new ArrayList<>();
-        for (Element spawnable :
-            XMLUtils.getChildren(
-                element, "item")) { // TODO Add more types of spawnables once entity parser is built
-          ItemStack stack = kitParser.parseItem(spawnable, false);
-          SpawnableItem item = new SpawnableItem(stack, numericID);
+        for (Element itemEl : XMLUtils.getChildren(spawnerEl, "item")) {
+          ItemStack stack = kitParser.parseItem(itemEl, false);
+          SpawnableItem item = new SpawnableItem(stack, "spawner-" + numericId.get());
           objects.add(item);
         }
 
+        ImmutableList.Builder<PotionEffect> chBuilder = ImmutableList.builder();
```

Comment:
This is outside the inner loop for<potion>, that means, the effects are gonna carry over:

```xml
<spanwers>
  <spawner>
    <potion id="potion-1">
      <effect>A</effect>
    </potion>
    <potion id="potion-2">
      <effect>B</effect>
    </potion>
  </spawner>
</spanwers>
```
In this example, potion 1 would have effect A, and potion 2 would have both effect A and B